zipline/pipeline/filters/filter.py

```diff
@@ -533,3 +535,27 @@ class StaticAssets(StaticSids):
     def __new__(cls, assets):
         sids = frozenset(asset.sid for asset in assets)
         return super(StaticAssets, cls).__new__(cls, sids)
+
+
+class AllPresent(CustomFilter, SingleInputMixin, StandardOutputs):
+    """Pipeline filter indicating input term has data for a given window.
+    """
+
+    def _validate(self):
+
+        if isinstance(self.inputs[0], Filter):
+            raise TypeError(
+                "Input to filter `AllPresent` cannot be a Filter."
+            )
+
+        return super(AllPresent, self)._validate()
+
+    def compute(self, today, assets, out, value):
+        if isinstance(value, LabelArray):
+            np_all(~value.is_missing(), axis=0, out=out)
```

Comment:
Small perf note: I think we can avoid making an extra copy of the full window here by rewriting this expression as `~np.any(value.is_missing, ...)`.  Lifting the negation to the outer scope means that we only do the negation on the final reduced row, rather than on the whole window.

Reply:
👍